### PR TITLE
Link SPIRV-Cross instead of using SDL_LoadObject

### LINKED ISSUE #44: Use linking instead of SDL_LoadObject where possible
We recommend vendoring the dxcompiler dependency, but the SDL_LoadObject calls use strings like "libdxcompiler.so", so if the client doesn't explicitly set LD_LIBRARY_PATH the dlopen calls will search the system libraries instead. Is it bad practice to use a relative path for dlopen? I feel like this is surprising behavior, but I understand if this is an issue for package managers.

### PR DESCRIPTION
**Pull request by thatcosmonaut**
Partially resolves #44 

We can probably stop vendoring spirv_cross_c.h and spirv.h now that we're linking SPIRV-Cross directly.

### CODE REVIEW

**Comment by madebr** [comment]
I added spirv-cross as a git submodule instead of using cmake's [`FetchContent`](https://cmake.org/cmake/help/latest/module/FetchContent.html) to do this. I believe this keeps complexity low.

The following CMake options are relevant:
- `SDLGPUSHADERCROSS_SHARED`: build shared SDL3_gpu_sharedcross library
- `SDLGPUSHADERCROSS_STATIC`: build static SDL3_gpu_sharedcross library
- `SDLGPUSHADERCROSS_VENDORED`: use vendored spirv-cross
- `SDLGPUSHADERCROSS_LINK_STATIC`: link to static spirv-cross

I also added `-Werror` when building the `shadercross` cli util, which errors now.

**Comment by thatcosmonaut** [comment]
Thanks, I'll look into that. While I'm at it do you have any insight on the best way to link DirectXShaderCompiler?

**Comment by madebr** [comment]
> Thanks, I'll look into that. While I'm at it do you have any insight on the best way to link DirectXShaderCompiler?

`SDLGPUSHADERCROSS_VENDORED` assumes the sources of spirv-cross are already available (checked out as a git submodule).
I think we can do something similar here.

For dependencies that don't provide CMake support (ignore pkg-config because that is not available by default on Windows), I usually write custom `FindXXX.cmake` modules.
Building it as a subproject is probably out of the question (LLVM is not fast to build).

Let's create a (cmake) script that downloads the prebuilt .so/dll binaries from the release binaries to a known location (e.g. `external/dxcompiler`)
Then by building with `SDLGPUSHADERCROSS_VENDORED=ON`, we will look in `external/dxcompiler` for the binaries.
If `SDLGPUSHADERCROSS_VENDORED=OFF`, then CMake will look in the default locations (and fail if it cannot find any).

**Comment by madebr** [comment]
I created a poc at https://github.com/madebr/SDL_gpu_shadercross/commit/e0df3340d5b626e17efc4274251c3393ea7d4b12 if you want to try things out.

**Comment by thatcosmonaut** [comment]
Awesome, I will check that out shortly.

**Comment by thatcosmonaut** [comment]
-Werror issue is fixed, it looks like verifying CMake files fails for SLR though.

File: src/cli.c [hunk 1]
```diff
@@ -57,7 +57,7 @@ int main(int argc, char *argv[])
     bool spirvSource = false;
     ShaderCross_ShaderFormat destinationFormat = SHADERFORMAT_INVALID;
     SDL_ShaderCross_ShaderStage shaderStage = SDL_SHADERCROSS_SHADERSTAGE_VERTEX;
-    SDL_ShaderCross_ShaderModel shaderModel;
+    SDL_ShaderCross_ShaderModel shaderModel = 0;
```
Comment (madebr) [inline src/cli.c:60]: That's `SDL_SHADERCROSS_SHADERMODEL_5_0`.
Is this ok, or should the enum contain an "invalid" shadermodel?

Comment (thatcosmonaut) [reply src/cli.c:60]: I'll add an invalid entry.

Comment (madebr) [reply src/cli.c]: Reading over the code, won't this invalid shader model cause a null pointer dereference?

Comment (madebr) [reply src/cli.c]: Nevermind, it's ok.

Comment (thatcosmonaut) [reply src/cli.c]: I don't think so. It just gets passed as a parameter to TranspileHLSLFromSPIRV which sets a numeric value for SPIRV-Cross in an if-else. Are you perhaps confusing shader model for shader stage?

Comment (thatcosmonaut) [reply src/cli.c]: Oh, I see. We should null guard the output of that function.

Comment (madebr) [reply src/cli.c]: I'm following the `shaderModel` variable in `cli.c`.
When no `-m` is passed it is invalid.
Functions that use `shaderModel` will return `NULL`, which will be written to the output.

**Comment by madebr** [comment]
Final note, I simply added the current main branch of spirv-cross as a submodule.
For SDL_image and SDL_mixer we use a libsdl-org fork of a tagged release.

**Comment by thatcosmonaut** [comment]
I think it's safe to use spirv-cross main, they don't do ABI breaking changes and main branch is generally bugfixes and improvements.

**Comment by madebr** [comment]
Adding it as a submodule "freezes" it though. So you might want to regularly update it to get latest bugfixes.

**Pull request merged**